### PR TITLE
Add editor support instructions to README

### PR DESCRIPTION
**Pull request by mayhewluke**

### CODE REVIEW

**Comment by siemato** [comment]
I was doing some work on the readme as well (and ironically just finished adding a table of content when i saw your post). I'm still editing the readme itself though. #303 is the PR, if you want to have a look at an earlier version. I would appreciate it, if you keep the TOC out for now.

**Comment by mayhewluke** [comment]
@siemato I don't see a TOC in #303 - did you have the chance to push it yet? I'd be happy to coordinate the TOC in this PR with yours so the PRs don't conflict.

For context, @anmonteiro requested I add a TOC in this PR because the PR increases the number of headings in the readme to a point where a TOC would be helpful.

File: README.md [hunk 1]
```diff
@@ -63,7 +63,44 @@ Melange has good support for Nix:
...
+information about your project.
+
+To enable editor support via `ocaml-lsp`, add the following to your `esy.json`:
+
+```json
+// esy.json
+{
+  "devDependencies": {
+    "@opam/ocaml-lsp-server": "ocaml/ocaml-lsp:ocaml-lsp-server.opam#c275140",
+    "@opam/dot-merlin-reader": "4.2"
```
Comment (anmonteiro) [inline README.md:79]:
```suggestion
    "@opam/dot-merlin-reader": "*"
```

**Comment by anmonteiro** [comment]
Thanks!

**Pull request merged**